index.js

```diff
@@ -896,20 +896,17 @@ export const String = NullaryTypeWithUrl
 
 //# RegexFlags :: Type
 //.
-//. Type comprising the canonical RegExp flags:
-//.
-//.   - `''`
-//.   - `'g'`
-//.   - `'i'`
-//.   - `'m'`
-//.   - `'gi'`
-//.   - `'gm'`
-//.   - `'im'`
-//.   - `'gim'`
+//. Type comprising the canonical RegExp flags *recognized by the runtime*.
+//. Characters must be alphabetically ordered. Repeated characters are not
+//. permitted. Invalid combinations (such as `'uv'`) are not permitted.
 export const RegexFlags = NullaryTypeWithUrl
   ('RegexFlags')
   ([String])
-  (s => /^g?i?m?$/.test (s));
+  (s => {
+     let regex;
+     try { regex = globalThis.RegExp ('', s); } catch { return false; }
+     return regex.flags === s;
```

Comment:
[22.2.6.4: get RegExp.prototype.flags][1] specifies the order of characters, so this check ensures that the characters of `s` are in alphabetical order. :)


[1]: https://tc39.es/ecma262/multipage/text-processing.html#sec-get-regexp.prototype.flags

Reply:
Oh, haha, we commented about this at the same time

Reply:
My concern still stands, though: Should that check be there? Are users forced to supply the flags in the order specified?

Reply:
```js
> new RegExp('', 'mg').flags
'gm'
```

Like, why don't we consider `mg` valid, even though the runtime accepts it?

Reply:
> Are users forced to supply the flags in the order specified?

Yes. This preserves the current behaviour (the predicate currently uses `/^g?i?m?$/`). Would you like to remove this requirement?

Reply:
Shouldn't we instead just check that `regex.flags.length === s.length`?

Reply:
For the record: I am in favour of removing that requirement. I don't think it's Sanctuary's job to enforce the order of RegEx flags, and should be left to something like a linter.